### PR TITLE
3.0.11 Handle exception when signing a malformed requestTemplate

### PR DESCRIPTION
**Pull request by gnarea**
Also moved a couple of tests into the right `describe()`.

### CODE REVIEW

File: lib/passport-wsfed-saml2/samlp.js [hunk 1]
```diff
@@ -162,7 +162,11 @@ Samlp.prototype = {
         };
 
         sig.signingKey = options.signingKey.key;
-        sig.computeSignature(SAMLRequest, { location: { reference: "//*[local-name(.)='Issuer']", action: 'after' } }); // Signature element must be located after Issuer
+        try {
+          sig.computeSignature(SAMLRequest, { location: { reference: "//*[local-name(.)='Issuer']", action: 'after' } }); // Signature element must be located after Issuer
+        } catch (e) {
+          return callback(new Error(e));
```
Comment (gnarea) [inline lib/passport-wsfed-saml2/samlp.js]: `e` is a string, which is why I wanted to wrap it in an `Error`.

File: test/samlp.tests.js [hunk 2]
```diff
@@ -7,59 +7,6 @@ var Samlp   = require('../lib/passport-wsfed-saml2/samlp');
 var Saml    = require('../lib/passport-wsfed-saml2/saml').SAML;
 
 describe('samlp (unit tests)', function () {
-  describe('samlrequest with template', function() {
```
Comment (gnarea) [inline test/samlp.tests.js]: The two tests inside actually belong to the `describe()` for `getSamlRequestParams()`, which is why I moved them (see below).

File: test/samlp.tests.js [hunk 3]
```diff
@@ -541,6 +488,79 @@ describe('samlp (unit tests)', function () {
         done();
       });
     });
+
+    it('should use providername option', function(done) {
```
Comment (gnarea) [inline test/samlp.tests.js]: Moved unchanged.

File: test/samlp.tests.js [hunk 4]
```diff
@@ -541,6 +488,79 @@ describe('samlp (unit tests)', function () {
...
+            .to.equal('Some name');
+
+          done();
+        });
+      });
+    });
+
+    describe('signing', function () {
+      describe('HTTP-POST or HTTP-Redirect without deflate encoding', function () {
+        it('should error if the requestTemplate is malformed', function (done) {
```
Comment (gnarea) [inline test/samlp.tests.js]: This is the new test.

File: test/samlp.tests.js [hunk 5]
```diff
@@ -541,6 +488,79 @@ describe('samlp (unit tests)', function () {
...
+          };
+          this.samlp.getSamlRequestParams(options, function(err, result) {
+            expect(err).to.be.an.Error;
+            expect(err.message).to.equal('end tag name: samlp:AuthnRequest is not match the current start tagName:undefined');
+            expect(result).to.not.exist;
+            done();
+          });
+        });
+
+        it('should place the signature after the issuer', function(done){
```
Comment (gnarea) [inline test/samlp.tests.js]: And this was also moved unchanged. I was going to implement a new "happy" test, but then I found it was actually available, but in the wrong place.

**Comment by ziluvatar** [comment]
https://sentry.io/auth0/auth0-server/issues/585151105/

**Pull request merged**